### PR TITLE
Fix global variables support

### PR DESCRIPTION
**Pull request by nbdd0121**
Fix linker script and boot.S so globals can be used without tag violations.

### CODE REVIEW

File: sw/cheri/link.ld [hunk 1]
```diff
@@ -9,4 +9,43 @@ SECTIONS
 		*(.text .text.* .rodata .rodata.* .data.rel.ro);
 		. = ALIGN(4);
 	}
+
+    .data :
+    {
+        __DATA_BEGIN__ = .;
+        *(.data .data.*)
```
Comment (engdoreis) [inline sw/cheri/link.ld:16]: Can you add a comment saying that the cheri compiler expects all the data sections together?

File: sw/cheri/common/boot.S [hunk 2]
```diff
@@ -7,28 +7,66 @@
...
-	// This could be a relative jump, but I'd need to get the relocations right
-	// and we have 32 KiB of IROM so wasting a few bytes doesn't really matter.
-	auipcc           cra, 0
-	li               t0, 0x00140000
-	csetaddr         cra, cra, t0
-	cjr              cra
+	li               sp, 0x00120000
+	csetaddr         csp, cs0, sp
+
+	// Clear the revocation bitmap before entering C code.
```
Comment (engdoreis) [inline sw/cheri/common/boot.S:18]:
```suggestion
	// Clear the revocation bitmap before entering C/C++ code.
```

**Review by engdoreis — APPROVED** [review]

File: sw/cheri/common/boot.S [hunk 3]
```diff
@@ -7,28 +7,66 @@
...
-	auipcc           cra, 0
-	// Call the C++ entry point
-	la_abs           t0, entry_point
-	csetaddr         cra, cra, t0
-	// Store capability to stack
-	csc              cra, 0(csp)
-	// Load capability from stack
-	clc              cra, 0(csp)
-	// Clear capability from stack
-	csc              cnull, 0(csp)
```
Comment (marnovandermaas) [inline sw/cheri/common/boot.S:25]: Removing these also removes them from CHERI sanity test. We should keep it that test and remove from all others.

Comment (nbdd0121) [reply sw/cheri/common/boot.S:25]: Added it to CHERI sanity via inline assembly. PTAL

File: sw/cheri/checks/cheri_sanity.cc [hunk 4]
```diff
@@ -16,6 +16,14 @@ extern "C" uint32_t entry_point(void *rwRoot)
...
 	Capability<void> root{rwRoot};
 
+	asm volatile(
+	  // Store capability to stack
+	  "csc              cra, -4(csp)\n"
+	  // Load capability from stack
+	  "clc              cra, -4(csp)\n"
+	  // Clear capability from stack
+	  "csc              cnull, -4(csp)\n" ::
+	    : "memory");
```
Comment (nbdd0121) [inline sw/cheri/checks/cheri_sanity.cc:26]: This is clang-format's formatting decision, not mine :)

**Comment by engdoreis** [comment]
@marnovandermaas can we merge this PR?

**Pull request merged**